docs/docs/integrations/language-models/anthropic.md

```diff
@@ -112,6 +114,264 @@ by setting `toolChoice(ToolChoice)` or `toolChoiceName(String)`.
 By default, Anthropic Claude may use multiple tools to answer a user query,
 but you can disable [parallel tool](https://docs.anthropic.com/en/docs/agents-and-tools/tool-use/implement-tool-use#parallel-tool-use) by setting `disableParallelToolUse(true)`.
 
+## Server Tools
+
+Anthropic's [server tools](https://platform.claude.com/docs/en/agents-and-tools/tool-use/overview#server-tools)
+are supported via `serverTools` parameter, here is an example of using a [web search tool](https://platform.claude.com/docs/en/agents-and-tools/tool-use/web-search-tool):
+```java
+AnthropicServerTool webSearchTool = AnthropicServerTool.builder()
+        .type("web_search_20250305")
+        .name("web_search")
+        .addAttribute("max_uses", 5)
+        .addAttribute("allowed_domains", List.of("accuweather.com"))
+        .build();
+
+ChatModel model = AnthropicChatModel.builder()
+        .apiKey(System.getenv("ANTHROPIC_API_KEY"))
+        .modelName("claude-sonnet-4-5")
+        .serverTools(webSearchTool)
+        .logRequests(true)
+        .logResponses(true)
+        .build();
+
+String answer = model.chat("What is the weather in Munich?");
+```
+
+Tools specified via `serverTools` will be included in every request to the Anthropic API.
+
+## Tool Search Tool
+
+Anthropic's [tool search tool](https://platform.claude.com/docs/en/agents-and-tools/tool-use/tool-search-tool)
+is supported via `serverTools`, tool `metadata` and `toolMetadataKeysToSend` parameters.
+
+Here is an example when using high-level AI Service and `@Tool` APIs:
+
+```java
+AnthropicServerTool toolSearchTool = AnthropicServerTool.builder()
+        .type("tool_search_tool_regex_20251119")
+        .name("tool_search_tool_regex")
+        .build();
+
+class Tools {
+
+    @Tool(metadata = "{\"defer_loading\": true}")
+    String getWeather(String location) {
+        return "sunny";
+    }
+
+    @Tool
+    String getTime(String location) {
+        return "12:34:56";
+    }
+}
+
+ChatModel chatModel = AnthropicChatModel.builder()
+        .apiKey(System.getenv("ANTHROPIC_API_KEY"))
+        .modelName(CLAUDE_SONNET_4_5_20250929)
+        .beta("advanced-tool-use-2025-11-20")
+        .serverTools(toolSearchTool)
+        .toolMetadataKeysToSend("deferLoading") // need to specify it explicitly
```

Comment:
I don't know the spec so maybe this is correct, but the key used here "deferLoading" is different from the one declared at line 157 "defer_loading", is this right?